lib/nfc/nfc_worker.c

```diff
@@ -522,6 +542,18 @@ void nfc_worker_mf_classic_dict_attack(NfcWorker* nfc_worker) {
         return;
     }
 
+    // Clear found keys if the key cache is incorrect (key set as found but sector not read)
+    for(uint16_t sector = 0; sector < total_sectors; sector++) {
+        if(mf_classic_is_key_found(data, sector, MfClassicKeyA) &&
+           !mf_classic_is_sector_read(data, sector)) {
+            mf_classic_set_key_not_found(data, sector, MfClassicKeyA);
+        }
+        if(mf_classic_is_key_found(data, sector, MfClassicKeyB) &&
+           !mf_classic_is_sector_read(data, sector)) {
+            mf_classic_set_key_not_found(data, sector, MfClassicKeyB);
+        }
+    }
+
```

Comment:
I am not sure that it's correct. Imagine sector that can be read by key B, which is not in dictionary. Then will remove correctly found key A

Reply:
Solved